### DISCUSSION TITLE
Connection Label Width

### DISCUSSION THREAD

**Discussion by rokf** [terrastruct/d2#603] [Q&A] [answered]
Hello,

Is it possible to define some kind of connection label width or somehow break the labels into multiple lines? I'm trying to draw C4 software architecture diagrams in D2 where labels are supposed to be very verbose. I couldn't find anything in the documentation.

I could probably hack something together using text shapes as an alternative but I'd love it if this would be possible as well.

Thank you 🙂 

**Edit:** I've just noticed that using Latex gives us some additional options but it's not (yet) supported in labels - https://d2lang.com/tour/text#latex 🤔

**Comment by alixander** [comment, marked as answer]
yes, you can just use `\n`:

https://play.d2lang.com/?script=dM0xrgIxDATQ_p9itPUnB0jBCtFsi2i3MdjCEYqNQqJwfJT01PNmhnA44haxbC17QTJUFZy9lRp3u0SEEMgYXamCE0MF7APZutspYhPcKQvUs_xjUJNPRfZiyR6Dd3qDhTjMvaujq9gM_Clor_n4q7Quf98AAAD__w%3D%3D

**Reply by rokf** [reply]
Thank you!
